homeassistant/components/climate/const.py

```diff
@@ -42,31 +42,64 @@ class HVACMode(StrEnum):
 HVAC_MODE_FAN_ONLY = "fan_only"
 HVAC_MODES = [cls.value for cls in HVACMode]
 
-# No preset is active
-PRESET_NONE = "none"
 
-# Device is running an energy-saving mode
-PRESET_ECO = "eco"
+class PresetMode(StrEnum):
+    """Preset mode for climate devices."""
 
-# Device is in away mode
-PRESET_AWAY = "away"
+    # No preset is active
+    NONE = "none"
 
-# Device turn all valve full up
-PRESET_BOOST = "boost"
+    # Device is running an energy-saving mode
+    ECO = "eco"
 
-# Device is in comfort mode
-PRESET_COMFORT = "comfort"
+    # Device is in away mode
+    AWAY = "away"
 
-# Device is in home mode
-PRESET_HOME = "home"
+    # Device turn all valve full up
+    BOOST = "boost"
 
-# Device is prepared for sleep
-PRESET_SLEEP = "sleep"
+    # Device is in comfort mode
+    COMFORT = "comfort"
 
-# Device is reacting to activity (e.g. movement sensors)
+    # Device is in home mode
+    HOME = "home"
+
+    # Device is prepared for sleep
+    SLEEP = "sleep"
+
+    # Device is reacting to activity (e.g. movement sensors)
+    ACTIVITY = "activity"
+
+
+# These PRESET_* constants are deprecated as of Home Assistant 2022.10.
+# Please use the PresetMode enum instead.
+PRESET_NONE = "none"
+PRESET_ECO = "eco"
+PRESET_AWAY = "away"
+PRESET_BOOST = "boost"
+PRESET_COMFORT = "comfort"
+PRESET_HOME = "home"
+PRESET_SLEEP = "sleep"
 PRESET_ACTIVITY = "activity"
 
-# Possible fan state
+
+class FanMode(StrEnum):
```

Comment:
I'm not sure this makes sense as these fan modes are not the only ones allowed. We allow any string as fan mode.

Side note: If we would create the climate integration today we wouldn't have included a fan service at all as we have a separate fan integration that can handle that.

Reply:
I was just thinking the same, looking at some of the integrations.
Does it still make sense for `SwingMode`?

Reply:
I think we should limit the modes for swing mode. Please check if there are any integrations that doesn't keep to the documented modes.

Reply:
`demo` uses `self._swing_modes = ["Auto", "1", "2", "3", "Off"]`
`tado` uses "ON"/"OFF" strings

And these three use external settings:
- `daikin` uses `self._attr_swing_modes = self._api.device.swing_modes`
- `melcloud` uses `self._device.vane_vertical_positions`
- `sensibo` uses `self.device_data.swing_modes`

Reply:
I have reverted the new enums, including SwingMode as it doesn't look like all desired swing modes really fit in an enum.